aikit/models/ordinal.py

```diff
@@ -0,0 +1,364 @@
+        """ helper method to go from the probabilities that the target is stricly above something to the proba of each class """
+        
+        # For example, if we have 4 ordered classes 'A', 'B', 'C' and 'D'
+        #
+        # probas_over = [ proba(y > A), proba(y > B), proba(y > C)] . So a list of 3 (= 4 -1 ) elements
+        #        
+        # This corresponds to the probas of target above something
+        # probas_over[j] := P( Y > classes[j] ) #
+        
+        # To go back to P( Y == classes[j] ) I need to do
+        
+        # P( Y == classes[0]   ) = 1- P(Y > classes[0] )   # smallest target
+        # P( Y == classes[j]   ) = P( Y > classes[j-1] ) - P( Y > classes[j] )  # intermediate target
+        # P( Y == classes[J-1] ) = P( Y > classes[J-2] )   # highest target
+        
+        J = len(probas_over)
+        
+        classes_proba = []
+        classes_proba.append( 1- probas_over[0] )
+        for j in range(1, J):
+            classes_proba.append( probas_over[j-1] - probas_over[j] )
+        classes_proba.append(probas_over[J-1])
+        
+        probas = np.concatenate(classes_proba, axis=1)
+        
+        return probas
+
+
+    def predict_proba(self, X):
+        
+        # Call classifier
+        y_hat_proba = self.classifier.predict_proba(X)
+        
+
+        # Retrive the proba of 1 from proba matrix
+        probas_over = []
+        for proba, cl in zip(y_hat_proba, self.classifier.classes_):
+            if cl[0] == 1:
+                p = proba[:,0]
+            else:
+                p = proba[:,1] # should always be the case
+                
+            assert len(cl) == 2
+            assert cl.tolist() == [0,1]
+            assert proba.shape[1] == 2
+
+            probas_over.append(p[:, np.newaxis])
+        
+        # Now let's re-aggregate the proba of each class
+        probas = []
+        start_index = 0
+        for target_index, categories in enumerate(self._target_encoder.categories_):
+
+            end_index = start_index + len(categories) - 1 # start and end of current target
+
+            probas_over_for_target = probas_over[start_index:end_index]
+            p = self._aggregate_probas_over(probas_over_for_target)
+            probas.append(p)
+            
+            start_index = end_index
+            
+        if self._mono_target:
+            return probas[0]
+        else:
+            return probas
+        
+        
+    def predict(self, X):
+        probas = self.predict_proba(X)
+        
+        classes = self.classes_
+        
+        if self._mono_target:
+            y_hat = classes[probas.argmax(axis=1)]
+        
+        else:
+            
+            all_pred = [ c[p.argmax(axis=1)][:, np.newaxis] for p, c in zip(probas, classes)]
+            y_hat = np.concatenate(all_pred, axis=1)
+            
+        return y_hat.astype(self._target_dtype)
+
+
+
+class RegressorFromClassifier(BaseEstimator, RegressorMixin):
```

Comment:
I put this class here, because it's kind of the same idea... but it is not really an ordinal classification (this it is a regression)